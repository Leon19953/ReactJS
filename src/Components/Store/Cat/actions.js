import { GET_CAT_REQUEST, GET_CAT_SUCCESS, GET_CAT_ERROR } from "./constants";

export const getCatRequestAction = () => ({
  type: GET_CAT_REQUEST,
});

export const getCatSuccessAction = (catUrl) => ({
  type: GET_CAT_SUCCESS,
  payload: catUrl,
});

export const getCatErrortAction = () => ({
  type: GET_CAT_ERROR,
});

export const getCatActionWithThunk = () => async (dispatch, getState) => {
  dispatch(getCatRequestAction());
  try {
    const response = await fetch("https://aws.random.cat/meow");

    if (!response.ok) {
      throw new Error(`error getting data with status`);
    }
    const result = await response.json();
    dispatch(getCatSuccessAction(result.file));
  } catch (error) {
    dispatch(getCatErrortAction);
  }
};
