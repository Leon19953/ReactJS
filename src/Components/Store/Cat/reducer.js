import { GET_CAT_REQUEST, GET_CAT_SUCCESS, GET_CAT_ERROR } from "./constants";

const initialState = {
  catUrl: undefined,
  loading: true,
  error: false,
};

export const catReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_CAT_REQUEST: {
      return {
        ...state,
        loading: true,
        error: false,
      };
    }
    case GET_CAT_SUCCESS: {
      return {
        ...state,
        loading: false,
        catUrl: action.payload,
      };
    }
    case GET_CAT_ERROR: {
      return {
        ...state,
        loading: false,
        error: true,
      };
    }
    default:
      return state;
  }
};
