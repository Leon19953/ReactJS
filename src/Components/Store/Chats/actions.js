import { ADD_CHAT_ACTION, DELETE_CHAT_ACTION } from "./constants";

export const addChatAction = ({ newChat }) => ({
  type: ADD_CHAT_ACTION,
  payload: { newChat },
});

export const deleteChatAction = ({ chatId }) => ({
  type: DELETE_CHAT_ACTION,
  payload: { chatId },
});
