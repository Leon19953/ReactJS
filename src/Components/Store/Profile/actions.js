import { TOGGLE_CHECKBOX } from "./constants";

export const toggleCheckboxAction = () => ({
  type: TOGGLE_CHECKBOX,
});
