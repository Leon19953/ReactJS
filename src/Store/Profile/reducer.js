import { TOGGLE_CHECKBOX } from "./constants";

const initialState = {
  checked: false,
  name: "Alexey",
};

export const profileReducer = (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_CHECKBOX: {
      return {
        ...state,
        checked: !state.checked,
      };
    }
    default:
      return state;
  }
};
