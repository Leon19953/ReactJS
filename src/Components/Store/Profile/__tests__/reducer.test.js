import { toggleCheckboxAction } from "../actions";
import { profileReducer } from "../reducer";

describe("test profile reducer", () => {
  it("if toggleCheckboxAction", () => {
    const expected = { checked: true, name: "Alexey" };

    const received = profileReducer(undefined, toggleCheckboxAction());

    expect(received).toEqual(expected);
  });
});
