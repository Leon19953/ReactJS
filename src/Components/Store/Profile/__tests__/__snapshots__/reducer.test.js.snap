// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile reducer if toggleCheckboxAction 1`] = `
Object {
  "checked": true,
  "name": "Alexey",
}
`;
